rclcpp/test/test_time_source.cpp

```diff
@@ -73,12 +142,20 @@ void trigger_clock_changes(
     msg->clock.sec = i;
     msg->clock.nanosec = 1000;
     clock_pub->publish(msg);
-    executor.spin_once(1000000ns);
```

Comment:
Another way to make the test work better was to call executor.spin_once ten times in a row here.  Then you'd start seeing the expected time at the end of a test run